specification/resources/resource-manager/Microsoft.Resources/stable/2024-04-01/resources.json

```diff
@@ -0,0 +1,6839 @@
+          },
+          "description": "The properties."
+        }
+      },
+      "description": "Resource type managed by the resource provider."
+    },
+    "Provider": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provider ID."
+        },
+        "namespace": {
+          "type": "string",
+          "description": "The namespace of the resource provider."
+        },
+        "registrationState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The registration state of the resource provider."
+        },
+        "registrationPolicy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The registration policy of the resource provider."
+        },
+        "resourceTypes": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ProviderResourceType"
+          },
+          "x-ms-identifiers": [
+            "resourceType"
+          ],
+          "description": "The collection of provider resource types."
+        },
+        "providerAuthorizationConsentState": {
+          "type": "string",
+          "enum": [
+            "NotSpecified",
+            "Required",
+            "NotRequired",
+            "Consented"
+          ],
+          "description": "The provider authorization consent state.",
+          "x-ms-enum": {
+            "name": "ProviderAuthorizationConsentState",
+            "modelAsString": true
+          }
+        }
+      },
+      "description": "Resource provider information."
+    },
+    "BasicDependency": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The ID of the dependency."
+        },
+        "resourceType": {
+          "type": "string",
+          "description": "The dependency resource type."
+        },
+        "resourceName": {
+          "type": "string",
+          "description": "The dependency resource name."
+        }
+      },
+      "description": "Deployment dependency information."
+    },
+    "Dependency": {
+      "properties": {
+        "dependsOn": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BasicDependency"
+          },
+          "description": "The list of dependencies."
+        },
+        "id": {
+          "type": "string",
+          "description": "The ID of the dependency."
+        },
+        "resourceType": {
+          "type": "string",
+          "description": "The dependency resource type."
+        },
+        "resourceName": {
+          "type": "string",
+          "description": "The dependency resource name."
+        }
+      },
+      "description": "Deployment dependency information."
+    },
+    "PreflightWarningResponseInfo": {
+      "type": "object",
+      "properties": {
+        "code": {
```

Comment:
Can we think of any scenarios where making this extensible with levels (warning, error, info) might be useful?

That's a genuine question BTW, not trying to suggest this is something we should do.

Reply:
Can you define extensible with levels? Are you referring to inheritance? Or grouping?

Reply:
I'm wondering if something like the following would be helpful:
```json
[{
  "level": "Warning",
  "code": "Blah",
  "message": "..."
}, {
  "level": "Info",
  "code": "Blah",
  "message": "..."
}]
```

Reply:
Got it, let me check on that, I like the idea though.